rclcpp_components/src/node_main.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "class_loader/class_loader.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_components/node_factory.hpp"
+
+
+#define LINKTIME_COMPOSITION_LOGGER_NAME "node_main"
+
+int main(int argc, char * argv[])
+{
+  // Force flush of the stdout buffer.
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+
+  auto args = rclcpp::init_and_remove_ros_arguments(argc, argv);
+  rclcpp::Logger logger = rclcpp::get_logger(LINKTIME_COMPOSITION_LOGGER_NAME);
+  rclcpp::executors::SingleThreadedExecutor exec;
+  rclcpp::NodeOptions options;
+  options.arguments(args);
+  std::vector<class_loader::ClassLoader *> loaders;
+  std::vector<rclcpp_components::NodeInstanceWrapper> node_wrappers;
+
+  std::vector<std::string> libraries = {
+    // all classes from libraries linked by the linker (rather then dlopen)
+    // are registered under the library_path ""
+    "",
+  };
+  for (auto library : libraries) {
+    RCLCPP_INFO(logger, "Load library %s", library.c_str());
```

Comment:
I think this line and the line below about instantiating the class should be debug rather than info. That's just my opinion, but I think that it's really an implementation detail and will get spamy when you have lots of nodes executed in this fashion.